src/Microsoft.Performance.SDK/HashUtils.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Performance.SDK
+{
+    /// <summary>
+    ///     Provides static methods for calculating hash.
+    /// </summary>
+    public static class HashUtils
+    {
+        /// <summary>
+        ///     Calculates the hash of all files under the given directory including the file paths.
+        /// </summary>
+        /// <param name="directory">
+        ///     Path to the directory.
+        /// </param>
+        /// <returns>
+        ///     The combined MD5 hash of all files including the file paths.
+        /// </returns>
+        public static async Task<string> GetDirectoryHash(string directory)
+        {
+            Guard.NotNull(directory, nameof(directory));
+            Guard.IsTrue(Directory.Exists(directory), $"Directory {directory} doesn't exist.");
+
+            using (var md5 = MD5.Create())
```

Comment:
Do we want to use the SHA256 class instead of MD5?

Due to collision problems with MD5/SHA1, Microsoft recommends SHA256 or SHA512. Consider using the [SHA256](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.sha256?view=net-7.0) class or the [SHA512](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.sha512?view=net-7.0) class instead of the [MD5](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.md5?view=net-7.0) class. Use [MD5](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.md5?view=net-7.0) only for compatibility with legacy applications and data.

Reply:
Updated